app/controllers/api/v1/facts_controller.rb

```diff
@@ -0,0 +1,61 @@
+class Api::V1::FactsController < ApplicationController
+  include AuthenticationCheck
+
+  before_action :is_user_logged_in
+  before_action :set_fact, only: [:show, :update, :destroy]
+  before_action :check_access
+
+  # GET /members/:member_id/facts
+  def index
+    @member = Member.find(params[:member_id])
+    render json: @member.facts
+  end
+
+  # GET /members/:member_id/facts/:id
+  def show
+    render json: @fact if check_access
```

Comment:
You don't need check access here since you are already calling it in the before action.

Reply:
Removed check access